Libraries/Blob/RCTBlobManager.m

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 2015-2016 Silk Labs, Inc.
+ * All Rights Reserved.
+ */
+
+#import "RCTBlobManager.h"
+
+#import <React/RCTConvert.h>
+
+static NSString *const kBlobUriScheme = @"blob";
+
+@implementation RCTBlobManager
+{
+  NSMutableDictionary<NSString *, NSData *> *_blobs;
+  NSOperationQueue *_queue;
+}
+
+RCT_EXPORT_MODULE(BlobModule)
+
+- (NSDictionary<NSString *, id> *)constantsToExport
+{
+  return @{
+    @"BLOB_URI_SCHEME": kBlobUriScheme,
+    @"BLOB_URI_HOST": [NSNull null],
+  };
+}
+
+- (NSString *)store:(NSData *)data
+{
+  NSString *blobId = [NSUUID UUID].UUIDString;
+  [self store:data withId:blobId];
+  return blobId;
+}
+
+- (void)store:(NSData *)data withId:(NSString *)blobId
+{
+  if (!_blobs) {
+    _blobs = [NSMutableDictionary new];
+  }
+  _blobs[blobId] = data;
+}
+
+- (NSData *)resolve:(NSDictionary<NSString *, id> *)blob
+{
+  NSString *blobId = [RCTConvert NSString:blob[@"blobId"]];
+  NSNumber *offset = [RCTConvert NSNumber:blob[@"offset"]];
+  NSNumber *size = [RCTConvert NSNumber:blob[@"size"]];
+  return [self resolve:blobId
+                offset:offset ? [offset integerValue] : 0
+                  size:size ? [size integerValue] : -1];
+}
+
+- (NSData *)resolve:(NSString *)blobId offset:(NSInteger)offset size:(NSInteger)size
+{
+  NSData *data = _blobs[blobId];
+  if (!data) {
+    return nil;
+  }
+  if (offset != 0 || (size != -1 && size != data.length)) {
```

Comment:
I think it's just a convenience. If you will look at the caller, it's only `- (NSData *)resolve:(NSDictionary<NSString *, id> *)blob` that refers to it. And that method is just a proxy, that extracts arguments from a passed `dictionary` and applies default parameters. 

Since both `offset` and `size` can be optional, I cannot think of a better alternative to that.